homeassistant/components/fivem/sensor.py

```diff
@@ -0,0 +1,98 @@
+"""The FiveM sensor platform."""
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import FiveMDataUpdateCoordinator, FiveMEntity
+from .const import (
+    ATTR_PLAYERS_LIST,
+    ATTR_RESOURCES_LIST,
+    DOMAIN,
+    ICON_PLAYERS_MAX,
+    ICON_PLAYERS_ONLINE,
+    ICON_RESOURCES,
+    NAME_PLAYERS_MAX,
+    NAME_PLAYERS_ONLINE,
+    NAME_RESOURCES,
+    UNIT_PLAYERS_MAX,
+    UNIT_PLAYERS_ONLINE,
+    UNIT_RESOURCES,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the FiveM sensor platform."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    # Create entities list.
+    entities = [
+        FiveMPlayersOnlineSensor(coordinator),
+        FiveMPlayersMaxSensor(coordinator),
+        FiveMResourcesSensor(coordinator),
```

Comment:
Have you had a look at the `EntityDescription` as well? See tuya as an example https://github.com/home-assistant/core/blob/dev/homeassistant/components/tuya/sensor.py.

This would make it easier to create those sensors and when you use the entity description, you don't need to set icon, device class and other attrs directly yourselves. You can just pass the description. This will make your code more readable and more aligned to core.

Reply:
This makes the creation of sensors way easier! Thanks for the help in the right direction